### ISSUE TITLE
[Feature Request] Drop jplayer for Plyr

### ISSUE THREAD

**Issue by kuzi-moto** [ampache/ampache#2113]
**Describe what you don't like about Ampache, or what it is missing**

jPlayer has served Ampache well for quite some time. Unfortunately it appears to no longer be updated, with the last release nearly 5 years ago. Bugs will not be fixed, and no new features/performance improvements added.

For Ampache to remain modern, I think it would be worth considering switching the player to [Plyr](https://github.com/sampotts/plyr) which seems to be a decent replacement. Development is very active, supports the last 2 versions of modern browsers, uses simple markup, and is configurable.

I know this is probably pretty low on the priorities list since the current player still works and Ampache has plenty of other issues, but thought it would be worth having on the radar.

**Comment by AshotN** [comment]
On a similar note, why is Ampache using Aurora.JS? 

If it's just for FLAC support, which is what I used to use Aurora.JS for. Modern browsers support FLAC natively now, around 90% support. https://caniuse.com/#feat=flac

**Comment by lachlan-00** [comment]
I have it disabled in my server but it looks like it's just a choice of audio decoder.
![image](https://user-images.githubusercontent.com/1305249/67920620-9d391f00-fbf0-11e9-819e-bebddda25fe6.png)

used in the player for these formats
![image](https://user-images.githubusercontent.com/1305249/67920652-c659af80-fbf0-11e9-9f6a-602d5df450f4.png)

**Comment by kuzi-moto** [comment]
Looks like Plyr relies on the browser's implementation of the codec to play. Only one that is not supported is m4a. The other codecs have decent support. It would be nice if Ampache had some basic telemetry for stuff like this, to determine what formats are being used by what browsers.

**Comment by AshotN** [comment]
I think some browsers to support M4A. Is it possible to make it a separate plugin @lachlan-00? I think Ampache can really benefit from some modularity and code cleanup.

**Comment by Afterster** [comment]
I added Aurora.js at a time browser support for FLAC didn't existed. In fact audio html5 support was *new* and wasn't yet working properly. It is also why we kept flash fallback. Few years later JPlayer support was getting slower. Today JPlayer is not supported anymore and all browser correctly play almost all audio formats.

Aurora.js could be removed I believe, as well as we should move to a more modern player, like Plyr suggested here (deeper analyze required, better from several people).

**Comment by xekon** [comment]
If somebody creates a test build or pull request I would be happy to run it for a while and my users and myself would be happy to test it. Plyr looks very nice!

**Comment by kuzi-moto** [comment]
Hopefully it's not too hard to implement. I found the code related to the current player quite confusing to read.

**Comment by KaeTuuN** [comment]
Any status update here?

**Comment by kuzi-moto** [comment]
I don't think it has been looked into much yet. It will likely take a bit of effort due to how convoluted the existing code is.

**Comment by wagnered** [comment]
The new architecture might be an opportunity to convert to plyr.  After all, I think the complexity occurs with managing the associated playlist and other visuals. The emphasis of the design was on HTML video and vimeo but It seems to play satisfactory HTML audio ( mp3 and ogg).

**Comment by lachlan-00** [comment]
i've been looking at this again and it just isn't something i know how to replace

**Comment by AshotN** [comment]
This will be one of those things that would need extensive testing after changing. Different browsers, different OSs... etc

**Comment by usox** [comment]
Despite of the actual implementation: Maybe it's possible to add a toggle to swtich between jplayer and plyr (default jplayer). So the testing wouldn't block the implementation and we could define a long grace period (e.g. removal in ampache 6)

**Comment by wagnered** [comment]
Why not do some evaluating before deciding?  [Media element.js](http://www.mediaelementjs.com/) seems to be in active development and it also has the MIT license:

>  ...HTML5 audio and video library that creates a unified feel for media files (MP4, MP3, FLV), streaming content (HLS, M(PEG)-DASH, RTMP), and embeddable players like YouTube, Vimeo, Twitch, DailyMotion, Facebook, and SoundCloud.

MediaElementjs might seem like an overkill but would allow for future feature expansion via plugins, which also appear to be in active development.

**Comment by kuzi-moto** [comment]
I'm not sure Media element.js supports many audio formats than MP3: https://github.com/mediaelement/mediaelement/issues/2136

**Comment by wagnered** [comment]
HTML5(native mode)  specs (according to w3c.org) supports ogg, wav, webm.  Mediaelement supports mp4 (m4a, m4v). and FLV.  The Wikipedia article on [HTML5  audio](https://en.m.wikipedia.org/wiki/HTML5_audio) indicates that major browsers support mp3 and FLAC.

So it seems to me that other comparisons should be made like how customizable is each player. Plyr might have the lead, although mediaelement.js adds features/customization via plugins.

**Comment by kuzi-moto** [comment]
Some of the plugins for mediaelement do look interesting, and may add some useful functionality.

**Comment by usox** [comment]
The contributor activity could also be a decision helper.
https://github.com/mediaelement/mediaelement/graphs/contributors
https://github.com/sampotts/plyr/graphs/contributors

**Comment by kuzi-moto** [comment]
Was thinking about this issue recently and figured I'd change the title as some discussion around _what_ will replace jplayer is necessary first.

That being said I'm not enthusiastic about the plyr and media element within the last couple of years. Plyr hasn't had a release in nearly one year and both projects have contributions from only a couple of people.

I would like to put my vote towards [video.js](https://github.com/videojs/video.js) now. It has an extremely consistent number of contributions for the last 10+ years, and there are numerous different contributors within the last 6-12 months. According to the README, it's used on 700,000+ websites already and has good plugin support.

**Comment by mitchray** [comment]
That's due to Plyr being absorbed into [Vidstack](https://github.com/vidstack/player), which is itself a self-proclaimed modern alternative to video.js
